src/platform/Linux/SystemTimeSupport.cpp

```diff
@@ -52,17 +52,64 @@ Milliseconds64 ClockImpl::GetMonotonicMilliseconds64()
 
 CHIP_ERROR ClockImpl::GetClock_RealTime(Clock::Microseconds64 & aCurTime)
 {
-    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+    struct timeval tv;
+    if (gettimeofday(&tv, nullptr) != 0)
+    {
+        return CHIP_ERROR_POSIX(errno);
+    }
+    if (tv.tv_sec < CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD)
+    {
+        return CHIP_ERROR_REAL_TIME_NOT_SYNCED;
+    }
+    if (tv.tv_usec < 0)
+    {
+        return CHIP_ERROR_REAL_TIME_NOT_SYNCED;
+    }
+    static_assert(CHIP_SYSTEM_CONFIG_VALID_REAL_TIME_THRESHOLD >= 0, "We might be letting through negative tv_sec values!");
+    aCurTime = Clock::Microseconds64((static_cast<uint64_t>(tv.tv_sec) * UINT64_C(1000000)) + static_cast<uint64_t>(tv.tv_usec));
+    return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR ClockImpl::GetClock_RealTimeMS(Clock::Milliseconds64 & aCurTime)
 {
-    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
```

Comment:
consider: call `GetClock_RealTimeMS(Clock::Microseconds64&)` and `duration_cast` the result to milliseconds?

Reply:
Fixed GetClock_RealTimeMS to use simplified logic.